source/libs/wal/src/walMeta.c

```diff
@@ -35,113 +35,251 @@ int64_t FORCE_INLINE walGetCommittedVer(SWal* pWal) { return pWal->vers.commitVe
 
 int64_t FORCE_INLINE walGetAppliedVer(SWal* pWal) { return pWal->vers.appliedVer; }
 
-static FORCE_INLINE void walBuildMetaName(SWal* pWal, int metaVer, char* buf) {
-  sprintf(buf, "%s/meta-ver%d", pWal->path, metaVer);
+static FORCE_INLINE int walBuildMetaName(SWal* pWal, int metaVer, char* buf) {
+  return sprintf(buf, "%s/meta-ver%d", pWal->path, metaVer);
 }
 
-static FORCE_INLINE int64_t walScanLogGetLastVer(SWal* pWal) {
+static FORCE_INLINE int walBuildTmpMetaName(SWal* pWal, char* buf) {
+  return sprintf(buf, "%s/meta-ver.tmp", pWal->path);
+}
+
+static FORCE_INLINE int64_t walScanLogGetLastVer(SWal* pWal, int32_t fileIdx) {
```

Comment:
There are  three different types of WAL files, i) meta info, ii) log files; iii) idx files, which are required to be in consistent condition almost always, e.g. especially on conditions of system or app crashing, disk fault, or file corruptions. It is a complicated issue to deal with since the three different types of files cannot be written into disk   atomically, confronting with various kinds of system failures, faults, or crashing, which is inevitably resulted in inconsistent states of different types of them. So it is vital to resolve these inconsistent states of files after crashing or fault. 
The original routines of inconsistency resolution are still incompletely, some of which are already noted in scattered comments within, which are walScanLogGetLastVer, walCheckAndRepairMeta, and walCheckAndRepairIdx.  So, they are refactored, with adjustment of walSaveMeta fsync policies and so on, hoping to be able to handle various problems of WAL recovery in customer sites.